mono/mini/mini-exceptions.c

```diff
@@ -2898,6 +2910,17 @@ print_overflow_stack_frame (StackFrameInfo *frame, MonoContext *ctx, gpointer da
 void
 mono_handle_hard_stack_ovf (MonoJitTlsData *jit_tls, MonoJitInfo *ji, void *ctx, guint8* fault_addr)
 {
+#if defined(DISABLE_STACK_OVERFLOW_GUARD) && defined(MONO_ARCH_HAVE_SIGCTX_TO_MONOCTX)
+	MonoContext mctx;
+
+	/*
+	 * We don't do soft stack overflow handling. Do our best here.
+	 * FIXME Don't do it in signal handler
+	 */
+	mono_sigctx_to_monoctx (ctx, &mctx);
```

Comment:
Hard stack overflow should always abort. This will 99.999% lead to a secondary abort that will hide the actual issue.